feat: add start and stop functions to qBittorrent v5 #138

farzanshibu opened the pull request:
feat: add start and stop functions to qBittorrent v5

- Fixed issue #138
- Implemented start and stop function for qBittorrent v5

scttcper commented:
thanks. i think what's tough here is you need to know which version for things like adding a torrent in the new "stopped" state https://github.com/farzanshibu/qbittorrent/blob/master/src/qbittorrent.ts#L607-L609 and it might also need the new states added to the other normalized functions. I don't currently do version checks for users, but it seems like that might also need to be an option on the function or something

farzanshibu commented:
I implemented a method to check the qBittorrent version during login using the API. This allows the application to determine the version and adjust accordingly, ensuring proper handling of torrent states such as pause, stop, start, and resume.

- Added a private _isVersion5OrHigher flag to track version compatibility
- Adjusted pauseTorrent(), resumeTorrent(), addTorrent(), and addMagnet() to select the correct endpoints and parameters based on the version detected.

scttcper commented:
Sorry for the delay i do plan on merging this
